src/authtoken/test.py

```diff
@@ -0,0 +1,226 @@
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+
+        auth = 'CardToken'
+        response = self.csrf_client.get(
+            self.path,
+            {'exp': stamp, 'account_id': str(account.pk)},
+            format='json',
+            HTTP_AUTHORIZATION=auth
+        )
+        self.assertEqual(response.status_code, status.HTTP_401_UNAUTHORIZED)
+
+        auth = 'CardToken {0} {1}'.format(token.decode()[:10], token.decode())
+        response = self.csrf_client.get(
+            self.path,
+            {'exp': stamp, 'account_id': str(account.pk)},
+            format='json',
+            HTTP_AUTHORIZATION=auth
+        )
+        self.assertEqual(response.status_code, status.HTTP_401_UNAUTHORIZED)
+
+
+@override_settings(ROOT_URLCONF='authtoken.test', SECRET_WEBTOKEN='secret')
+class CreateFooCardTokenTests(TestCase):
+    enc = 'HS256'
+    path = '/auth-token/'
+
+    def setUp(self):
+        self.csrf_client = APIClient(enforce_csrf_checks=True)
+
+    def test_create_new_account_token(self):
+        card = CardFactory()
+        api_token = Token.objects.create(key='abc123')
+        auth = 'Token {0}'.format(api_token)
+        response = self.csrf_client.post(
+            self.path,
+            {'number': card.number},
+            format='json',
+            HTTP_AUTHORIZATION=auth
+        )
+
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        self.assertIn('card_token', response.data.keys())
+
+        token = response.data.get('card_token')
+        self.assertIsNotNone(token)
+        self.assertTrue(len(token) > 0)
+
+        try:
+            decoded = jwt.decode(token, 'secret', algorithm=self.enc)
+        except jwt.exceptions.InvalidTokenError as e:
+            self.fail(e)
+
+        account = Account.objects.all().first()
+        self.assertEqual(decoded.get('account_id'), str(account.pk))
+        self.assertIn('exp', decoded.keys())
+
+    def test_create_token_for_invalid_account(self):
+        api_token = Token.objects.create(key='abc123')
+        auth = 'Token {0}'.format(api_token)
+        response = self.csrf_client.post(
+            self.path,
+            {'number': '123456'},
+            format='json',
+            HTTP_AUTHORIZATION=auth
+        )
+
+        self.assertEqual(response.status_code, status.HTTP_400_BAD_REQUEST)
+        self.assertIn('number', response.data.keys())
+
+    def test_create_token_with_incorrect_data(self):
+        api_token = Token.objects.create(key='abc123')
+        auth = 'Token {0}'.format(api_token)
+        response = self.csrf_client.post(
+            self.path,
+            {'sumtinwong': '123456'},
+            format='json',
+            HTTP_AUTHORIZATION=auth
+        )
+
+        self.assertEqual(response.status_code, status.HTTP_400_BAD_REQUEST)
+        self.assertIn('number', response.data.keys())
+
+        response = self.csrf_client.post(
```

Comment:
Wouldn't it be enough with one test here? You are doing three different tests with different incorrect data and I'm not really sure it is necessary. They basically all test the same kind of behaviour - that the endpoint can differentiate correct from incorrect data.